iOS/Flows/Conversation/Cells/ConversationMessageCellDatasource.swift

```diff
@@ -85,3 +85,16 @@ extension ConversationMessageCellDataSource {
         }
     }
 }
+
+// MARK: - TimelineCollectionViewLayoutDataSource
+
+extension ConversationMessageCellDataSource: TimelineCollectionViewLayoutDataSource {
+
+    func getMessage(at indexPath: IndexPath) -> Messageable? {
+        guard let item = self.itemIdentifier(for: indexPath) else { return nil }
+        let messageController = ChatClient.shared.messageController(cid: item.channelID,
+                                                                    messageId: item.messageID)
+
+        return messageController.message
+    }
+}
```

Comment:
I made a data source delegate protocol for the time machine layout so it doesn't need an explicit reference to the specific datasource type.